core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -206,6 +191,89 @@ public void testFailedCommit() throws Exception {
     Assertions.assertThat(manifests).as("Should contain 0 Avro manifest files").isEmpty();
   }
 
+  @Test
+  public void testCommitFailedBeforeChangeVersionHint() throws Exception {
+    // create v1-format table,for easy testing
+    Map<String,String> confMap = new HashMap<>();
+    confMap.put(TableProperties.FORMAT_VERSION,"1");
+    String newLocation = tableLocation+"v1"+File.separator;
+    Table v1Table = TABLES.create(SCHEMA, SPEC,confMap,newLocation);
+    // Submit a new file to test the functionality.
+    v1Table.newFastAppend().appendFile(FILE_A).commit();
```

Comment:
This is fine but it looks like we also upgrade the version of the table below? Not sure why. Also shouldn't we do this after we capture the metadata of the initial table state?

Reply:
Because Iceberg currently creates V2 tables by default. For testing purposes, I created the V1 table manually and tested a commit to prove that it works.